app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -161,6 +162,87 @@ public void onError(Object errorMessage) {
+            if (user != null && user.getLinks() != null && user.getLinks().getYonaMessages() != null
+                    && !TextUtils.isEmpty(user.getLinks().getYonaMessages().getHref())) {
+
+                notificationNetwork.getNextSetOfMessagesFromURL(urlForMessagesFetch, YonaApplication.getEventChangeManager().getSharedPreference().getYonaPassword(), isUnreadStatus, new DataLoadListener() {
+                    @Override
+                    public void onDataLoad(Object result) {
+                        if (listener != null) {
+                            if (result instanceof YonaMessages) {
+                                YonaMessages resultYonaMessagesObj = (YonaMessages) result;
+                                if (resultYonaMessagesObj != null && resultYonaMessagesObj.getEmbedded() != null) {
+                                    Embedded embedded = resultYonaMessagesObj.getEmbedded();
+                                    List<YonaMessage> listMessages = embedded.getYonaMessages();
+                                    boolean isAnyProcessed = false;
+                                    SimpleDateFormat sdf = new SimpleDateFormat(AppConstant.YONA_LONG_DATE_FORMAT, Locale.getDefault());
+                                    for (YonaMessage message : listMessages) {
+                                        //update enum
+                                        if (message.getStatus() != null) {
+                                            message.setNotificationMessageEnum(NotificationMessageEnum.getNotificationMessageEnum(message.getType(), message.getStatus()));
+                                        } else if (message.getDropBuddyReason() != null) {
+                                            message.setNotificationMessageEnum(NotificationMessageEnum.getNotificationMessageEnum(message.getType(), message.getDropBuddyReason()));
+                                        } else {
+                                            message.setNotificationMessageEnum(NotificationMessageEnum.getNotificationMessageEnum(message.getType(), message.getChange()));
+                                        }
+                                        if (message.getLinks() != null && message.getLinks().getYonaUser() != null
+                                                && !TextUtils.isEmpty(message.getLinks().getYonaUser().getHref())) {
+                                            if (message.getEmbedded() == null) {
+                                                message.setEmbedded(new Embedded());
+                                            }
+                                            message.getEmbedded().setYonaUser(getYonaUser(message.getLinks().getYonaUser().getHref()));
+                                        }
+                                        String uploadDate = "";
+
+                                        String createdTime = message.getCreationTime();
+                                        try {
```

Comment:
[Our coding conventions](https://wiki.yona.nu/display/DEV/Code) (which are new to you) demand that a try/catch always spans the entire method body (that's one more reason to keep the method short (<= 20 lines).

Reply:
Will break into functions.

Reply:
I will remove all unnecessary nested try/catch's